extension-qr-code/src/org/eclipse/birt/extension/qrcode/util/SwtGraphicsUtil.java

```diff
@@ -0,0 +1,122 @@
+/*******************************************************************************
+ * Copyright (c) 2022 Henning von Bargen
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * https://www.eclipse.org/legal/epl-2.0/.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ *
+ * Contributors:
+ *  Henning von Bargen - initial API and implementation
+ *******************************************************************************/
+
+package org.eclipse.birt.extension.qrcode.util;
+
+import java.util.HashMap;
+
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.graphics.GC;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.swt.graphics.ImageData;
+import org.eclipse.swt.graphics.PaletteData;
+import org.eclipse.swt.graphics.RGB;
+import org.eclipse.swt.widgets.Display;
+
+import com.google.zxing.BarcodeFormat;
+import com.google.zxing.EncodeHintType;
+import com.google.zxing.common.BitArray;
+import com.google.zxing.common.BitMatrix;
+import com.google.zxing.qrcode.QRCodeWriter;
+import com.google.zxing.qrcode.decoder.ErrorCorrectionLevel;
+/**
+ * SwtGraphicsUtil
+ */
+public class SwtGraphicsUtil {
+
+	public static Image createQRCodeImage(String text, int dotsWidth, int dotsHeight, String encoding,
```

Comment:
"create" method should never return `null`, it should throw exception instead if creation failed